khook/x86/hook.c

```diff
@@ -86,22 +176,31 @@ static inline void khook_arch_sm_init_one(khook_t *hook) {
 
 	memcpy(stub->orig, hook->target.addr, stub->nbytes);
 	x86_put_jmp(stub->orig + stub->nbytes, stub->orig + stub->nbytes, hook->target.addr + stub->nbytes);
-	kernel_write_enter();
```

Comment:
I'd suggest to change `kernel_write_enter()` to `kernel_write_enter(ptr, len)` and implement PTE logic there, based on compile-time macro value.

Reply:
Okey, got it. Will be done